File: LeetCode/704_Binary_Search.js
var search = function(nums, target) {
    nums = nums.sort(function(a, b) {return a - b})
    var l = 0, r = nums.length - 1;

    while (l <= r) {
        m = parseInt((l+r)/2)
        if (nums[m] < target) {
            l = m + 1
        } else if (nums[m] > target) {
            r = m - 1
        } else {
            return m
        }
    }
    return -1
}


var nums = [-1,0,3,5,9,12], target = 9
console.log(search(nums, target))

File: LeetCode/704_Binary_Search.py
def search(nums, target):
    nums.sort()
    print(nums)
    l, r = 0, len(nums) - 1
    
    while l <= r:
        m = ((l + r) // 2)
        print(m)
        if nums[m] > target:
            r = m - 1
        elif nums[m] < target:
            l = m + 1
        else:
            return m
    return -1





nums = [-1,0,3,5,9,12] 
target = 9
nums = [5] 
target = 5

print(search(nums, target))